src/legacy/ui/public/agg_types/controls/order_by.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { EuiFormRow, EuiSelect } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { AggParamEditorProps } from 'ui/vis/editors/default';
+import { safeMakeLabel, isCompatibleAggregation } from '../agg_utils';
+
+const aggFilter = [
+  '!top_hits',
+  '!percentiles',
+  '!median',
+  '!std_dev',
+  '!derivative',
+  '!moving_avg',
+  '!serial_diff',
+  '!cumulative_sum',
+  '!avg_bucket',
+  '!max_bucket',
+  '!min_bucket',
+  '!sum_bucket',
+];
+const isCompatibleAgg = isCompatibleAggregation(aggFilter);
+
+function OrderByParamEditor({
+  agg,
+  value,
+  showValidation,
+  setValue,
+  setValidity,
+  setTouched,
+  responseValueAggs,
+}: AggParamEditorProps<string>) {
+  const label = i18n.translate('common.ui.aggTypes.orderAgg.orderByLabel', {
+    defaultMessage: 'Order by',
+  });
+  const isValid = !!value;
+
+  useEffect(
+    () => {
+      setValidity(isValid);
```

Comment:
Are there any bad sideeffects of calling `setValidity`, or couldn't we just always call `setValidity(isValid)` without any `useEffect` hook around it?

Reply:
It produces side effect, because `setValidity` calls state update: https://github.com/elastic/kibana/pull/39502/files?file-filters%5B%5D=.tsx#diff-10db5f379a5d1774e97d202e364f6226R175


Without `useEffect` hook it breaks (`Uncaught Invariant Violation: Maximum update depth exceeded.`)